src/azul/files.py

```diff
@@ -15,3 +16,18 @@ def write_file_atomically(path, mode=0o644):
     except BaseException:
         os.unlink(temp_path)
         raise
+
+
+def file_sha1(path):
+    """
+    >>> file_sha1('/dev/null')
```

Comment:
I could see why testing on `__file__` is not a good idea but collisions weren't my motivation for my comment so I don't really understand your [response](https://github.com/DataBiosphere/azul/pull/1655#discussion_r406519238) to it.

Can you think of an easy way to test this with a file size of 1M -1, 1M and 1M+1? How about os.urandom and BytesIO? That would still be a one-liner AFAICT.

Reply:
`files.py` should now be included in the doctests suite.

Reply:
> I don't really understand your response to it.

I think you could exploit collisions in a hashing function in order to get it to contain it's own hash. I'm not sure of this anyway, but that was my thinking.